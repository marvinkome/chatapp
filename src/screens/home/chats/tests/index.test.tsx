import * as React from 'react';
import { shallow } from 'enzyme';
import { Chats } from '../index';
import View from '../view';
import ListItem from '../view/listing';

describe('<Chat/> listing tab', () => {
    const lists = [
        {
            name: 'Larry John',
            text: 'I am mocked :)',
            image: '',
            timestamp: '12:47 AM'
        }
    ];

    // snapshot
    it('renders correctly', () => {
        const app = shallow(<Chats />);

        expect(app).toMatchSnapshot();
    });

    // list all chats
    it('should list all chats', () => {
        const wrapper = shallow(<View lists={lists} />);
        const list = wrapper.find('List');

        expect(list).toMatchSnapshot();
        expect(list.props().children).toHaveLength(1);
    });

    describe('<ListItem/>', () => {
        // a list item should show the number of unread post
        it('shows the number of unread messages', () => {
            const wrapper = shallow(
                <ListItem listItem={lists[0]} typing={false} />
            );
            const badgeProp: { element: any } = wrapper.shallow().prop('badge');

            expect(badgeProp.element).toMatchSnapshot();
            expect(badgeProp.element.props.children[1].props.value).toBe(3);
        });

        // when user is typing change the text
        it('should show typing message when some user is typing', () => {
            const wrapper = shallow(
                <ListItem listItem={lists[0]} typing={true} />
            );
            const text = wrapper.shallow().prop('subtitle');

            expect(text).toBe('typing...');
        });

        // show number of notifications in tab header
        // pressing a list item should go to the chat page
    });
});
