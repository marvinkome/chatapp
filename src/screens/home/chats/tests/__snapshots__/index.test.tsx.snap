// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat/> listing tab <ListItem/> shows the number of unread messages 1`] = `
<Component
  style={
    Object {
      "paddingLeft": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "flex-end",
        "color": "hsl(0, 0%, 45%)",
        "fontFamily": "Karla",
        "fontSize": 13,
        "marginRight": 10,
        "paddingBottom": 4,
      }
    }
  >
    12:47 AM
  </Text>
  <Badge
    containerStyle={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#42ED86",
        "marginRight": 15,
        "padding": 7,
      }
    }
    textStyle={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontFamily": "Karla",
        "fontSize": 13,
      }
    }
  />
</Component>
`;

exports[`<Chat/> listing tab renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chats />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "query": Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "user",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "groups",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "first",
                                  },
                                  "value": Object {
                                    "kind": "IntValue",
                                    "value": "1",
                                  },
                                },
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "sort",
                                  },
                                  "value": Object {
                                    "kind": "BooleanValue",
                                    "value": true,
                                  },
                                },
                              ],
                              "directives": Array [
                                Object {
                                  "arguments": Array [
                                    Object {
                                      "kind": "Argument",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "key",
                                      },
                                      "value": Object {
                                        "block": false,
                                        "kind": "StringValue",
                                        "value": "messages",
                                      },
                                    },
                                  ],
                                  "kind": "Directive",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "connection",
                                  },
                                },
                              ],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "messages",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "message",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "timestamp",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "role",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 377,
          "start": 0,
        },
      },
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "query": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": undefined,
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "user",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "groups",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [
                                  Object {
                                    "kind": "Argument",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "first",
                                    },
                                    "value": Object {
                                      "kind": "IntValue",
                                      "value": "1",
                                    },
                                  },
                                  Object {
                                    "kind": "Argument",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "sort",
                                    },
                                    "value": Object {
                                      "kind": "BooleanValue",
                                      "value": true,
                                    },
                                  },
                                ],
                                "directives": Array [
                                  Object {
                                    "arguments": Array [
                                      Object {
                                        "kind": "Argument",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "key",
                                        },
                                        "value": Object {
                                          "block": false,
                                          "kind": "StringValue",
                                          "value": "messages",
                                        },
                                      },
                                    ],
                                    "kind": "Directive",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "connection",
                                    },
                                  },
                                ],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "messages",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "message",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "timestamp",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "role",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "name",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 377,
            "start": 0,
          },
        },
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Chat/> listing tab should list all chats 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <MainView
      lists={
        Array [
          Object {
            "image": "",
            "name": "Larry John",
            "text": "I am mocked :)",
            "timestamp": "12:47 AM",
          },
        ]
      }
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <List
          containerStyle={
            Object {
              "borderTopWidth": 0,
              "marginTop": 13,
            }
          }
        >
          <Listing
            listItem={
              Object {
                "image": "",
                "name": "Larry John",
                "text": "I am mocked :)",
                "timestamp": "12:47 AM",
              }
            }
            typing={false}
          />
        </List>,
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Listing
              listItem={
                Object {
                  "image": "",
                  "name": "Larry John",
                  "text": "I am mocked :)",
                  "timestamp": "12:47 AM",
                }
              }
              typing={false}
            />,
          ],
          "containerStyle": Object {
            "borderTopWidth": 0,
            "marginTop": 13,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "listItem": Object {
                "image": "",
                "name": "Larry John",
                "text": "I am mocked :)",
                "timestamp": "12:47 AM",
              },
              "typing": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <List
            containerStyle={
              Object {
                "borderTopWidth": 0,
                "marginTop": 13,
              }
            }
          >
            <Listing
              listItem={
                Object {
                  "image": "",
                  "name": "Larry John",
                  "text": "I am mocked :)",
                  "timestamp": "12:47 AM",
                }
              }
              typing={false}
            />
          </List>,
          "style": Object {
            "backgroundColor": "#fff",
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Listing
                listItem={
                  Object {
                    "image": "",
                    "name": "Larry John",
                    "text": "I am mocked :)",
                    "timestamp": "12:47 AM",
                  }
                }
                typing={false}
              />,
            ],
            "containerStyle": Object {
              "borderTopWidth": 0,
              "marginTop": 13,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "listItem": Object {
                  "image": "",
                  "name": "Larry John",
                  "text": "I am mocked :)",
                  "timestamp": "12:47 AM",
                },
                "typing": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Listing
          listItem={
            Object {
              "image": "",
              "name": "Larry John",
              "text": "I am mocked :)",
              "timestamp": "12:47 AM",
            }
          }
          typing={false}
        />,
      ],
      "containerStyle": Object {
        "borderTopWidth": 0,
        "marginTop": 13,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "listItem": Object {
            "image": "",
            "name": "Larry John",
            "text": "I am mocked :)",
            "timestamp": "12:47 AM",
          },
          "typing": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Listing
            listItem={
              Object {
                "image": "",
                "name": "Larry John",
                "text": "I am mocked :)",
                "timestamp": "12:47 AM",
              }
            }
            typing={false}
          />,
        ],
        "containerStyle": Object {
          "borderTopWidth": 0,
          "marginTop": 13,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "listItem": Object {
              "image": "",
              "name": "Larry John",
              "text": "I am mocked :)",
              "timestamp": "12:47 AM",
            },
            "typing": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
